HealthcareSystem/Backend/Repository/PatientRepository/MySqlPatientRepository/MySqlActivePatientRepository.cs

```diff
@@ -42,10 +42,23 @@ public List<Patient> GetAllPatients()
         {
             return _context.Patients.ToList();
         }
+
         public Patient GetPatientByJmbg(string jmbg)
         {
-            return _context.Patients.Find(jmbg);
+            Patient patient;
+            try
+            {
+                patient = _context.Patients.Find(jmbg);
+            }
+            catch (Exception)
+            {
+                throw new DatabaseException("The database connection is down.");
+            }
+            if (patient == null)
+                throw new NotFoundException("Patient doesn't exist in database.");
+            return patient;
```

Comment:
This is better, but your Controller method isn't catching it, if this actually happens you'll get server failure. Add another catch block for NotFoundException in the method.

Reply:
Fixed in be98915f313550c2f0c40f3f02babbc905a76b36